src/MultiOwnerLightAccount.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.23;
+
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {BaseAccount} from "account-abstraction/core/BaseAccount.sol";
+import {SIG_VALIDATION_FAILED} from "account-abstraction/core/Helpers.sol";
+import {IEntryPoint} from "account-abstraction/interfaces/IEntryPoint.sol";
+import {PackedUserOperation} from "account-abstraction/interfaces/PackedUserOperation.sol";
+import {TokenCallbackHandler} from "account-abstraction/samples/callback/TokenCallbackHandler.sol";
+import {CastLib} from "modular-account/helpers/CastLib.sol";
+import {SetValue} from "modular-account/libraries/Constants.sol";
+import {LinkedListSet, LinkedListSetLib} from "modular-account/libraries/LinkedListSetLib.sol";
+
+import {CustomSlotInitializable} from "./CustomSlotInitializable.sol";
+
+/// @title A simple ERC-4337 compatible smart contract account with one or more designated owner accounts.
+/// @dev Like LightAccount, but multiple owners are supported. The account is initialized with a list of owners,
+/// and the `updateOwners` method can be used to add or remove owners.
+contract MultiOwnerLightAccount is
+    BaseAccount,
+    TokenCallbackHandler,
+    UUPSUpgradeable,
+    CustomSlotInitializable,
+    IERC1271
+{
+    using ECDSA for bytes32;
+    using MessageHashUtils for bytes32;
+    using LinkedListSetLib for LinkedListSet;
+    using CastLib for address;
+    using CastLib for SetValue[];
+
+    // keccak256(abi.encode(uint256(keccak256("multi_owner_light_account_v1.storage")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 internal constant _STORAGE_POSITION = 0x0eb5184329babcda7203727c83eff940fb292fc735f61720a6182b755bf7f900;
+    // keccak256(abi.encode(uint256(keccak256("multi_owner_light_account_v1.initializable")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 internal constant _INITIALIZABLE_STORAGE_POSITION =
+        0xaa296a366a62f6551d3ddfceae892d1791068a359a0d3461aab99dfc6c5fd700;
+    // bytes4(keccak256("isValidSignature(bytes32,bytes)"))
+    bytes4 internal constant _1271_MAGIC_VALUE = 0x1626ba7e;
+    IEntryPoint private immutable _ENTRY_POINT;
+    bytes32 private constant _DOMAIN_SEPARATOR_TYPEHASH =
+        keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
+    bytes32 private constant _LA_MSG_TYPEHASH = keccak256("MultiOwnerLightAccountMessage(bytes message)");
+    bytes32 private constant _NAME_HASH = keccak256("MultiOwnerLightAccount");
+    bytes32 private constant _VERSION_HASH = keccak256("1");
```

Comment:
hm... I think this might make a difference if the LA is upgraded between the single owner and the multi owner versions. If they're the same, it would allow a "replay" of previous signatures using only one of the owners, but that might be OK? Since 1271 signatures expect the caller to perform the replay protections.

I think the bigger concerns are around SDK usage, where the signing requests might need to be aware of the version, which could cause debugging issues if signatures aren't working due to using the wrong LA version 712 wrapper.

I'm leaning towards making it the same for ease of SDK integration, but could be convinced otherwise.